### PR TITLE
EE: Add CompileExpressions() method to compile multiple expressions to a single assembly

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

**Comment by cston** [comment]
@kevinh-ms, @plnelson, @jacdavis please review.
cc @jaredpar, @ManishJayaswal, @jinujoseph.

**Comment by jacdavis** [comment]
@cston The API looks like what I expected.

**Comment by cston** [comment]
cc @ivanbasov

File: src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs [hunk 1]
```diff
@@ -492,47 +522,51 @@ private static LocalAndMethod MakeLocalAndMethod(LocalSymbol local, MethodSymbol
         private EEMethodSymbol GetLocalMethod(EENamedTypeSymbol container, string methodName, string localName, int localIndex)
         {
             var syntax = SyntaxFactory.IdentifierName(localName);
-            return this.CreateMethod(container, methodName, syntax, (EEMethodSymbol method, DiagnosticBag diagnostics, out ImmutableArray<LocalSymbol> declaredLocals) =>
+            return this.CreateMethod(container, methodName, syntax, (EEMethodSymbol method, DiagnosticBag diagnostics, out ImmutableArray<LocalSymbol> declaredLocals, out ResultProperties properties) =>
```
Comment (KevinH-MS) [inline src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CompilationContext.cs]: consider: "unused" or "_*" for unused lambda parameters (here and other invocations)

**Comment by KevinH-MS** [comment]
General question:  Is it possible (or important) to associate a particular diagnostic message with a particular input expression?

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/CompileExpressionsTests.cs [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.ExpressionEvaluator.UnitTests;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator.UnitTests
+{
+    public class CompileExpressionsTests : ExpressionCompilerTestBase
+    {
+        [Fact]
```
Comment (KevinH-MS) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/CompileExpressionsTests.cs]: consider: tests for void expressions and assignments/expression statements

File: src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs [hunk 3]
```diff
@@ -218,15 +218,66 @@ internal sealed class EvaluationContext : EvaluationContextBase
...
+                    }
+                }
+            }
+            if (assembly == null)
+            {
+                methodTokens = ImmutableArray<int>.Empty;
+                errorMessages = ImmutableArray.CreateRange(
+                    diagnostics.AsEnumerable().
+                        Where(d => d.Severity == DiagnosticSeverity.Error).
+                        Select(d => GetErrorMessage(d, CSharpDiagnosticFormatter.Instance, preferredUICulture: null)));
```
Comment (KevinH-MS) [inline src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EvaluationContext.cs]: Did we ever make the general change to drop the line/col information from the error messages?  I thought we did, and I was just looking at the test baselines and thinking it's probably superfluous.

**Comment by cston** [comment]
@KevinH-MS, ideally error messages should be associated with particular expressions. However, some errors, such as those detected during emit of the single assembly, cannot be easily associated with a particular expression. I've added a comment to `CompileExpressions` for now.

File: src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/CompileExpressionsTests.cs (hunk 2 above)
Comment (cston) [inline src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/CompileExpressionsTests.cs]: Added tests.

**Pull request merged**